src/frontend/src/optimizer/plan_node/generic/agg.rs

```diff
@@ -348,7 +348,8 @@ impl<PlanRef: stream::StreamPlanRef> Agg<PlanRef> {
         let in_dist_key = self.input.distribution().dist_column_indices().to_vec();
 
         let gen_materialized_input_state = |sort_keys: Vec<(OrderType, usize)>,
-                                            include_keys: Vec<usize>|
+                                            include_keys: Vec<usize>,
+                                            distinct_key: Option<usize>|
```

Comment:
I suggest make this `|sort_keys: Vec<(OrderType, usize)>, extra_keys: Vec<usize>, include_cols: Vec<usize>|`, where `sort_keys` are keys with specific order types, `extra_keys` are keys to ensure each row unique, `include_cols` are extra columns should be included in the state table.

Reply:
Good idea. Updated